twyla/service/event.py

```diff
@@ -9,21 +9,27 @@
 
 import twyla.service.jsontool as jsontool
 
+def split_event_name(event_name: str):
+    assert "." in event_name, "Event names should be of format domain.event_name"
+    return event_name.split('.', 1)
+
 
 class Event:
-    def __init__(self, channel, body, envelope, name):
+
+    def __init__(self, channel, body, envelope):
         self.channel = channel
         self.body = body
         self.envelope = envelope
-        self.name = name
-
-    def payload(self):
-        """
-        The payload method is where the deserialization and validation of the
-        event body happens. It returns an EventPayload object. The schema for
-        deserialization and validation is loaded from a central schema service.
-        """
-        return EventPayload.from_json(self.body)
+        self.payload = None
+        self.event_name = None
+        self.domain = None
+        self.event_type = None
```

Comment:
Just a question out of interest. Last week, we had a discussion regarding `None` and how empty values of types should be used instead of it. Is there any particular reason then why you are using `None` here?

Reply:
Good point. It's usually a good idea to set empty values for attributes if they are set later, this value is then None. Setting it to emtpy strings etc. here would not be a good idea, because other code would happily pass along empty strings instead of throwing an error. This usage is suboptimal, but kinda unavoidable given the way Python works.

Reply:
Ah I see -- that make a lot of sense. Thanks!